frontend/src/lib/components/lemonToast.tsx

```diff
@@ -0,0 +1,86 @@
+import React from 'react'
+import { toast, ToastOptions } from 'react-toastify'
+import { IconCheckmark, IconErrorOutline, IconInfo, IconWarningAmber } from './icons'
+import { LemonButton } from './LemonButton'
+
+interface ToastButton {
+    label: string
+    action: () => void
+}
+
+interface ToastOptionsWithButton extends ToastOptions {
+    button?: ToastButton
+}
+
+const GET_HELP_BUTTON: ToastButton = {
+    label: 'Get help',
+    action: () => {
+        window.open('https://posthog.com/support?utm_medium=in-product&utm_campaign=error-toast', '_blank')
+    },
+}
+
+interface ToastContentProps {
+    message: string | JSX.Element
+    button?: ToastButton
+    id?: number | string
+}
+
+function ToastContent({ message, button, id }: ToastContentProps): JSX.Element {
+    return (
+        <div className="flex-center">
+            <span style={{ flexGrow: 1 }}>{message}</span>
+            {button && (
+                <LemonButton
+                    onClick={() => {
+                        button.action()
+                        toast.dismiss(id)
+                    }}
+                    type="secondary"
+                    compact
+                >
+                    {button.label}
+                </LemonButton>
+            )}
+        </div>
+    )
+}
+
+function ensureToastId(toastOptions: ToastOptions): void {
+    if (!toastOptions.toastId) {
+        toastOptions.toastId = `lemon-${Math.round(Math.random() * 10000000)}`
+    }
+}
```

Comment:
Nit: I'd add an explicit comment here why it's fine to mutate the object (`...toastOptions`), as opposed to all the other places where we default to disallowing this kind of behaviour. Alternatively, rewriting this in a functional way (returns a new object) won't really cost us much performance.

Reply:
Fair, converted it to the functional way